benchmarks/ltx.js

```diff
@@ -0,0 +1,79 @@
+'use strict'
+
+/*
+  benchmark the speed of the different methods to create elements
+  Not all tests are equally functional but it gives a good idea of what to expect from
+  the different techniques.
+ */
+
+var benchmark = require('benchmark')
+var ltx = require('../index')
+var createElement = ltx.createElement
+var tag = ltx.tag
+var Element = ltx.Element
+var parse = ltx.parse
+
+var XML = [
+  '<message to="foo@bar" from="bar@foo" type="chat" id="foobar">',
+  '<body>Where there is love there is life.</body>',
+  '</message>'
+].join('')
+var el = parse(XML)
+
+var suite = new benchmark.Suite('ltx')
+
+suite.add('tag with template literal', function () {
+  tag`
+    <message to="${'foo@bar'}" from="${'bar@foo'}" type="${'chat'}" id="${'foobar'}">
+      <body>${'Where there is love there is life.'}</body>
+    </message>
+  `
+})
+
+suite.add('tag with direct call', function () {
+  tag(
+    [
+      '\n  <message to="',
+      '" from="',
+      '" type="',
+      '" id="',
+      '">\n    <body>',
+      '</body>\n  </message>\n'
+    ],
+    'foo@bar',
+    'bar@foo',
+    'chat',
+    'foobar',
+    'Where there is love there is life.'
+  )
+})
+
+suite.add('serialize and parse', function () {
+  parse(el.toString())
+})
+
+suite.add('parse', function () {
+  parse(XML)
+})
+
+suite.add('serialize', function () {
+  el.toString()
+})
+
+suite.add('createElement (jsx)', function () {
+  createElement(
+    'message', {to: 'foo@bar', from: 'bar@foo', type: 'chat', id: 'foobar'},
+    createElement('body', null, 'Where there is love there is life.')
+  )
+})
+
+suite.add('clone', function () {
+  el.clone()
+})
+
+suite.add('Element', function () {
+  new Element('message', {to: 'foo@bar', from: 'bar@foo', type: 'chat', id: 'foobar'})
+    .c('body').t('Where there is love there is life.').root()
+})
+
+module.exports = suite
```

Comment:
sneak peak

```
➜  ltx git:(moar-benchmarks) ✗ node benchmarks/ltx.js 
tag with template literal x 32,518 ops/sec ±2.47% (64 runs sampled)
tag with direct call x 32,588 ops/sec ±4.75% (70 runs sampled)
serialize and parse x 43,464 ops/sec ±3.35% (67 runs sampled)
parse x 52,801 ops/sec ±4.59% (66 runs sampled)
serialize x 304,275 ops/sec ±7.16% (64 runs sampled)
createElement (jsx) x 349,110 ops/sec ±5.61% (70 runs sampled)
clone x 925,548 ops/sec ±3.54% (69 runs sampled)
Element x 1,094,036 ops/sec ±3.32% (68 runs sampled)
Fastest is Element
```

some comments
* 1M ops/s for `new Element` :heart: 
* serialize and parse is faster than parse, weird, probably some v8 optimization
* tag really performs badly, probably due to the use of the arguments object